editor/debugger/editor_debugger_node.cpp

```diff
@@ -81,6 +80,8 @@ EditorDebuggerNode::EditorDebuggerNode() {
 	editor->get_undo_redo()->set_method_notify_callback(_method_changeds, this);
 	editor->get_undo_redo()->set_property_notify_callback(_property_changeds, this);
 	editor->get_pause_button()->connect("pressed", callable_mp(this, &EditorDebuggerNode::_paused));
+
+	tabs->connect("tab_changed", callable_mp(this, &EditorDebuggerNode::_debugger_changed));
```

Comment:
Could you add an explanatory comment in front of each `connect` call moved to explain why it needs to happen last?

Reply:
Ok, something like:

> // This connect signal must be on the bottom of this constructor because the "tab_changed" signal is emitted by "add_child" functions and "tab_changed" signals handlers need the object to be ready, otherwise it will crash
> tabs->connect("tab_changed", callable_mp(this, 
 EditorDebuggerNode::_debugger_changed));

Please, correct me

Is understood **"tab_changed" signals handlers**?

Reply:
Now that I read the full explanation, I wonder if this wouldn't be sufficient to fix the bugs?
```
diff --git a/scene/gui/tab_container.cpp b/scene/gui/tab_container.cpp
index 240e07aba3..6290d364fc 100644
--- a/scene/gui/tab_container.cpp
+++ b/scene/gui/tab_container.cpp
@@ -538,7 +538,7 @@ void TabContainer::add_child_notify(Node *p_child) {
 
 	update();
 	p_child->connect("renamed", callable_mp(this, &TabContainer::_child_renamed_callback));
-	if (first)
+	if (first && is_inside_tree())
 		emit_signal("tab_changed", current);
 }
```

At least locally it fixes the crashes when loading a project in the editor.

Reply:
Ok, I read a lot of code to understand the behavior (I started with Godot this week)

And you're right, simples, and better because we get similar behavior from last signal system when we call "addChild".